### PR TITLE
Tiled Galleries: fix dynamic property deprecation notices

### PR DESCRIPTION
**Pull request by jeherve**
Follow-up from 30786

## Proposed changes:

Fix more deprecation notices introduced with PHP 8.2:

```
PHP Deprecated:  Creation of dynamic property Jetpack_Tiled_Gallery_Rectangular_Item::$img_src is deprecated in modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php on line 56
PHP Deprecated:  Creation of dynamic property Jetpack_Tiled_Gallery_Rectangular_Item::$img_src_grayscale is deprecated in modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php on line 134
PHP Deprecated:  Creation of dynamic property Jetpack_Tiled_Gallery_Rectangular_Item::$size is deprecated in modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php on line 136
```

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Primary issue: #27927

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

On a site connected to WordPress.com:

* Go to Posts > Add New and add a Tiled Gallery block.
* Ensure such galleries are displayed properly on the frontend.

### CODE REVIEW

File: projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php [hunk 1]
```diff
@@ -122,6 +163,20 @@ public function large_file() {
  * Tiled gallery rectangular item class.
  */
 class Jetpack_Tiled_Gallery_Rectangular_Item extends Jetpack_Tiled_Gallery_Item { // phpcs:ignore Generic.Files.OneObjectStructurePerFile.MultipleFound, Generic.Classes.OpeningBraceSameLine.ContentAfterBrace
+	/**
+	 * The image URL, in grayscale.
+	 *
+	 * @var string
+	 */
+	public $img_src_grayscale;
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php]: Looking at the one use of this property, it looks like it probably belongs on the parent class alongside `grayscale`.

https://github.com/Automattic/jetpack/blob/9a3a10ad9562ee5649d39ad815a52a49fb258783/projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/templates/partials/item.php#L43-L51

File: projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php [hunk 2]
```diff
@@ -122,6 +163,20 @@ public function large_file() {
...
+
+	/**
+	 * The image size.
+	 *
+	 * @var string
+	 */
+	public $size;
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php]: Similarly, the way the property is used it probably belongs on the parent class  as `string|null`.

https://github.com/Automattic/jetpack/blob/9a3a10ad9562ee5649d39ad815a52a49fb258783/projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/templates/partials/item.php#L18-L19

File: projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php [hunk 3]
```diff
@@ -145,6 +200,13 @@ public function __construct( $attachment_image, $needs_attachment_link, $graysca
  * Tiled gallery square item class.
  */
 class Jetpack_Tiled_Gallery_Square_Item extends Jetpack_Tiled_Gallery_Item { // phpcs:ignore Generic.Files.OneObjectStructurePerFile.MultipleFound, Generic.Classes.OpeningBraceSameLine.ContentAfterBrace
+	/**
+	 * The image URL, in grayscale.
+	 *
+	 * @var string
+	 */
+	public $img_src_grayscale;
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php]: (see above)

**Comment by jeherve** [comment]
@anomiex Thanks for the feedback. This should be addressed in b14c892f776917cfc7802aa07c81f27a0a61a90b.

**Review by anomiex — APPROVED** [review]

**Pull request merged**